main.go

```diff
@@ -201,18 +197,21 @@ func main() {
 				os.Exit(1)
 			}
 		} else {
-			setupLog.Info(fmt.Sprintf("Found operator namespace %s.", operatorNs))
-			hubCfg, err := common.LoadHubConfig(hubConfigSecretNs, hubConfigSecretName)
-			if err != nil {
-				setupLog.Error(err, "Unable to load hub kubeconfig, setting up leaseUpdater anyway")
-			}
-
+			setupLog.V(2).Info("Got operator namespace", "Namespace", operatorNs)
 			setupLog.Info("Starting lease controller to report status")
+
 			leaseUpdater := lease.NewLeaseUpdater(
 				generatedClient,
 				"iam-policy-controller",
 				operatorNs,
-			).WithHubLeaseConfig(hubCfg, clusterName)
+			)
+
+			hubCfg, err := clientcmd.BuildConfigFromFlags("", hubConfigPath)
+			if err != nil {
+				setupLog.Error(err, "Could not load hub config, lease updater not set with config")
+			} else {
+				leaseUpdater = leaseUpdater.WithHubLeaseConfig(hubCfg, clusterName)
+			}
 
 			go leaseUpdater.Start(context.TODO())
```

Comment:
Should this run when `err` is not `nil`?

Reply:
I think so -the HubLeaseConfig is only used when the cluster the controller is in doesn't support the Lease API: https://pkg.go.dev/open-cluster-management.io/addon-framework@v0.2.0/pkg/lease#LeaseUpdater
> 	// WithHubLeaseConfig sets the lease config on hub cluster. It allows LeaseUpdater to create/update
>	// addon lease on hub cluster when resource 'Lease' is not available on managed cluster.

So on clusters with the lease API, we can start the leaseUpdater and it will work fine. In the case where the lease API is not available *and* the hub cfg can't be loaded, the controller can still function, it just won't report its health correctly. So I think this is better than erroring out and stopping the controller.